src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationOverlay.tsx

```diff
@@ -239,6 +258,12 @@ export const CreateOrganizationOverlay: FC = () => {
             dispatch(notify(quartzOrgQuotaReached()))
           }
 
+          event(CreateOrgOverlayEvent.OrgCreationSuccess, multiOrgTag, {
+            createdOrgId: newOrg.id,
+            createdOrgName: newOrg.name,
+            currentOrgId,
+          })
```

Comment:
This method being called again and being essentially the same as `handleEventing` is another clue that maybe the abraction of that method is in the wrong place. If you take my suggestion above, this code can be changed to just `handleEventing()` (or whatever the name of that method becomes). Which to me is an indicator that maybe the abstractions should be moved around.

In my experience, the wrong abstraction leads to hard abstractions to create, while the right abstraction makes things a lot easier - if we can move the abstraction around and in doing so remove some duplicated code while making the method a little simpler, that is usually a good indicator that we're on the right track. This is a hueristic I use when trying to design software systems.

Reply:
This sounds good to me - but here I would implement the same comment I mentioned above re: we should have the same names for the keys of the object literal sent by the event - 'old' versus 'current' for org id/name - whether the event is coming from the Create Organization overlay or elsewhere in the app. That'll make for less confusion on the analytics side having to tailor SQL queries differently for different events.

I would use the same key names as the old events (rather than changing the old events' key names) to avoid disturbing the existing data that already exists re: usage of the switch-org components.